src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -420,42 +428,48 @@ case class WdlBinding(ast: Ast, source: WdlSource, importResolver: ImportResolve
           }
 
           /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
-          val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
+          val memberAccesses = WdlBinding.findTopLevelMemberAccesses(inputKv._2.ast)
+          /*val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
             case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
-          }
-          memberAccesses.keys map validateMemberAccess
+          }*/
+          memberAccesses.map(partialEvalMemberAccess).map{y => y.get}
         }
       }
     }
   }
 
   /* Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time */
-  def validateMemberAccess(ast: Ast): Any = {
+  def partialEvalMemberAccess(ast: Ast): Try[Call] = {
+    def callFromName(name: String): Try[Call] = {
+      val callList = workflows.head.calls.filter {call =>
```

Comment:
Is `workflows` guaranteed to have a `head`? Also, you should be able to replace the `call => call` with `_`